eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/ExternalDependencyExposedCheck.java

```diff
@@ -198,7 +191,23 @@ private boolean isValidClassDependency(String typeName) {
         }
 
         final String qualifiedName = simpleClassNameToQualifiedNameMap.get(typeName);
-        return VALID_DEPENDENCY_SET.stream()
-            .anyMatch(validPackageName -> qualifiedName.startsWith(validPackageName));
+
+        if ("com.".regionMatches(0, qualifiedName, 0, 4)) {
+            if ("azure.".regionMatches(0, qualifiedName, 4, 6)) {
+                return true;
+            } else if ("generic.".regionMatches(0, qualifiedName, 4, 8)) {
+                return true;
+            } else {
+                return false;
+            }
+        } else if ("java.".regionMatches(0, qualifiedName, 0, 5)) {
+            return true;
+        } else if ("reactor.".regionMatches(0, qualifiedName, 0, 8)) {
+            return true;
+        } else if ("org.reactivestreams.".regionMatches(0, qualifiedName, 0, 20)) {
+            return true;
```

Comment:
We would need tooling updates once we shift the project to it owns repo and tooling to not have reactor and other azure-core references.